unity/Assets/Scripts/AgentManager.cs

```diff
@@ -728,147 +704,184 @@ private void addImageSynthesisImage(List<KeyValuePair<string, byte[]>> payload,
     }
 
 
-	public IEnumerator EmitFrame() {
+    private bool canEmit() {
+        bool emit = true;
+        foreach (BaseFPSAgentController agent in this.agents) {
+            if (agent.agentState != AgentState.Emit) {
+                emit = false;
+                break;
+            }
+        }
 
-		frameCounter += 1;
+        return this.agentState == AgentState.Emit && emit;
+    }
 
-		bool shouldRender = this.renderImage && serverSideScreenshot;
 
-		if (shouldRender) {
-			// we should only read the screen buffer after rendering is complete
+	public IEnumerator EmitFrame() {
+        while (true) 
+        {
+            bool shouldRender = this.renderImage && serverSideScreenshot;
 			yield return new WaitForEndOfFrame();
-		}
-
-        MultiAgentMetadata multiMeta = new MultiAgentMetadata ();
-		
-
-		ThirdPartyCameraMetadata[] cameraMetadata = new ThirdPartyCameraMetadata[this.thirdPartyCameras.Count];
-        List<KeyValuePair<string, byte[]>> renderPayload = new List<KeyValuePair<string, byte[]>>();
-        createPayload(multiMeta, cameraMetadata, renderPayload, shouldRender);
 
+            frameCounter += 1;
+            if (this.agentState == AgentState.ActionComplete) {
+                this.agentState = AgentState.Emit;
```

Comment:
Related to the commend above, It is a little bit confusing to me that we have both `agentState` in the `AgentManager` Class and in the `BaseFPSAgentController` class, since `AgentManager` just manages multiple agents and `AgentState` sounds like an Agent level property not a manager one?

Reply:
I agree it is a little weird in terms of naming, but we need to have state on the agentManager itself because it has actions like AddThirtpartyCamera, Initialize, etc.  So I figured it should go through the same state transitions.  I suppose it could be called agentManagerState to be less confusing?